### PR TITLE
ocp4 - AC-2 (11) - mark as not applicable

### PR DESCRIPTION
**Pull request by itewk**
peer reviewed by:
* kevindoingbizas

### CODE REVIEW

**Comment by redhatrises** [comment]
I would make the argument that this is applicable but is part of IdP configuration. Let's use some of the same language that we did previously in AC-2(1).

**Comment by itewk** [comment]
@mradecker and @kevindoingbizas can you take a look at @redhatrises comment and weigh in?

**Comment by mradecker** [comment]
The supplemental guidance for this control enhancement is:

This control enhancement helps to enforce the principle of least
privilege, increase user accountability, and enable more effective account monitoring. Such
monitoring includes, for example, alerts generated if the account is used outside of specified
parameters. Organizations can describe the specific conditions or circumstances under which
system accounts can be used, for example, by restricting usage to certain days of the week,
time of day, or specific durations of time.

Can OCP 4 address this by itself?  If not, what can we couple it with to address this control enhancement?  Also, AC-2(11) is only applicable to systems categorized as High.

**Comment by itewk** [comment]
@mradecker @redhatrises so what do we want to update the state / narrative to given all that?

**Comment by redhatrises** [comment]
Basically my viewpoint is that we should mark this as `complete` and update the description to say:

```
This functionality would fall to third
party software such as an Identity Provider (IdP),
Active Directory (MSAD), LDAP, etc.
To configure an IdP, see https://docs.openshift.com/container-platform/latest/authentication/identity_providers/configuring-ldap-identity-provider.html#configuring-ldap-identity-provider
```
because many IdPs provide technical capabilities to set when a user can login, what day, etc.

**Comment by itewk** [comment]
@redhatrises maybe add another part that includes something about monitoring the aduit logs for alerts on those actions as per @mradecker supplimental information?

```
This functionality would fall to third
party software such as an Identity Provider (IdP),
Active Directory (MSAD), LDAP, etc.
To configure an IdP, see https://docs.openshift.com/container-platform/latest/authentication/identity_providers/configuring-ldap-identity-provider.html#configuring-ldap-identity-provider

For active alerting on auditable events then then OpneShift audit log system would need to be enabled and the OpenShift Elasticsearch, Kibarana, and FluentD (EFK) stack configured to alert on the organizational specified audit events, or a third party tool configured to consume the OpenShift audit log and alert on the organizational defined audit events.
```

**Comment by shawndwells** [comment]
> Can OCP 4 address this by itself? If not, what can we couple it with to address this control enhancement?


No, OCP cannot address by itself. When we triaged this control with the OCP team there was general agreement that meeting AC-2(11) would fall onto third party IdM systems. So still applicable, but the language could be as @redhatrises suggested.

> Also, AC-2(11) is only applicable to systems categorized as High.

Often find ourselves in situations where FISMA High controls are selected in ICD 503 and DHS 4300A, which together represent a large portion of Federal customers.

**Comment by mradecker** [comment]
@shawndwells Thanks for the information.  Based on your input, I would recommend marking this complete and include the information @redhatrises suggested.

**Comment by itewk** [comment]
@shawndwells @mradecker @redhatrises updated PR. others feel free to continue to update.

File: openshift-container-platform-4/policies/AC-Access_Control/component.yaml [hunk 1]
```diff
@@ -232,16 +232,23 @@
...
   narrative:
     - text: |-
-        This control will only be partially satisfied when OpenShift is
-        configured to connect to an Identity Provider (IdP).
-
-        More work is being done to address this control. Engineering
-        progress can be tracked via:
-
-        https://issues.redhat.com/browse/CMP-210
+       This functionality would fall to third
```
Comment (shawndwells) [inline openshift-container-platform-4/policies/AC-Access_Control/component.yaml]:
```suggestion
       This functionality is the responsibility of third
```

File: openshift-container-platform-4/policies/AC-Access_Control/component.yaml [hunk 2]
```diff
@@ -232,16 +232,23 @@
...
+       party software such as an Identity Provider (IdP),
```
Comment (shawndwells) [inline openshift-container-platform-4/policies/AC-Access_Control/component.yaml]:
```suggestion
       party software such as Red Hat Identity Management (Red Hat IdM),
```

Comment (shawndwells) [reply openshift-container-platform-4/policies/AC-Access_Control/component.yaml]: The nuance here is "Identity Provider (IdP)" being a category, whereas Red Hat IdM, MSAD, LDAP, etc are examples of appropriate 3rd party software.

File: openshift-container-platform-4/policies/AC-Access_Control/component.yaml [hunk 3]
```diff
@@ -232,16 +232,23 @@
...
+       Active Directory (MSAD), LDAP, etc.
+
+       To configure an IdP, see https://docs.openshift.com/container-platform/latest/authentication/identity_providers/configuring-ldap-identity-provider.html#configuring-ldap-identity-provider
```
Comment (shawndwells) [inline openshift-container-platform-4/policies/AC-Access_Control/component.yaml]:
```suggestion
       To configure an Identity Provider (IdP), refer to the "Authentication" chapter of the OpenShift Container Platform documentation at https://docs.openshift.com/container-platform/latest/authentication/understanding-authentication.html
```

Comment (shawndwells) [reply openshift-container-platform-4/policies/AC-Access_Control/component.yaml]: The current URL is specific to LDAP. Should the generic "authentication" chapter URL be used instead?

File: openshift-container-platform-4/policies/AC-Access_Control/component.yaml [hunk 4]
```diff
@@ -232,16 +232,23 @@
...
+
+       For active alerting on auditable events then then
+       OpneShift audit log system (https://docs.openshift.com/container-platform/latest/nodes/nodes/nodes-nodes-audit-log.html)
```
Comment (shawndwells) [inline openshift-container-platform-4/policies/AC-Access_Control/component.yaml]: Small typos

File: openshift-container-platform-4/policies/AC-Access_Control/component.yaml [hunk 5]
```diff
@@ -232,16 +232,23 @@
...
+       would need to be enabled
+       and the OpenShift Elasticsearch, Kibarana,
+       and FluentD (EFK) stack configured to alert on the
+       organizational specified audit events, or a third party
+       tool configured to consume the OpenShift audit log and
+       alert on the organizational defined audit events.
```
Comment (shawndwells) [inline openshift-container-platform-4/policies/AC-Access_Control/component.yaml]: The control language does not mention active alerting, only enforcement of circumstances/usage conditions:

AC-2(11) | ACCOUNT MANAGEMENT \| USAGE CONDITIONSThe information system enforces [Assignment: organization-defined circumstances and/or usage conditions] for [Assignment: organization-defined information system accounts]. Supplemental Guidance: Organizations can describe the specific conditions or circumstances under which information system accounts can be used, for example, by restricting usage to certain days of the week, time of day, or specific durations of time.
-- | --

Is the language around OpenShift alerting needed?


``

Comment (mradecker) [reply openshift-container-platform-4/policies/AC-Access_Control/component.yaml]: @shawndwells We can take it out.  I used the supplemental guidance for Rev 5 but Rev 4 doesn't have that verbiage.  If you're using Rev 4, then go with what you have.

File: openshift-container-platform-4/policies/AC-Access_Control/component.yaml [hunk 6]
```diff
@@ -232,16 +232,23 @@
...
-        progress can be tracked via:
-
-        https://issues.redhat.com/browse/CMP-210
+       This functionality is the responsibility of third
+       party software such as Red Hat Identity Management (Red Hat IdM),
+       Active Directory (MSAD), LDAP, etc.
+
+       To configure an Identity Provider (IdP), refer to the "Authentication" chapter of the OpenShift Container Platform documentation at https://docs.openshift.com/container-platform/latest/authentication/understanding-authentication.html
+
+       For active alerting on auditable events then then
```
Comment (shawndwells) [inline openshift-container-platform-4/policies/AC-Access_Control/component.yaml]:
```suggestion
       For active alerting on auditable events then the
```

File: openshift-container-platform-4/policies/AC-Access_Control/component.yaml [hunk 7]
```diff
@@ -232,16 +232,23 @@
...
+       OpneShift audit log system (https://docs.openshift.com/container-platform/latest/nodes/nodes/nodes-nodes-audit-log.html)
```
Comment (shawndwells) [inline openshift-container-platform-4/policies/AC-Access_Control/component.yaml]:
```suggestion
       OpenShift audit log system (https://docs.openshift.com/container-platform/latest/nodes/nodes/nodes-nodes-audit-log.html)
```

File: openshift-container-platform-4/policies/AC-Access_Control/component.yaml (hunk 5 above)
Comment (shawndwells) [reply openshift-container-platform-4/policies/AC-Access_Control/component.yaml]: @mradecker ok, took it out. 

Updating everything to rev5 (once finalized) will be suuuuuper fun.

**Comment by shawndwells** [comment]
Updated manually and merged. Thanks @itewk !

**Comment by itewk** [comment]
@shawndwells thanks.

**Pull request merged**